lib/ci/ci_result_parser.js

```diff
@@ -873,48 +910,52 @@ class NormalBuild extends Job {
     }
 
     const { result, runs, builtOn } = data;
+    //console.log(data);
     this.builtOn = builtOn;
 
     if (result !== FAILURE) {
       this.failures = [];
       return this.failures;
     }
 
-    if (!runs.length) {
-      this.failures = [
-        new BuildFailure(
-          { url: this.jobUrl, builtOn }, 'Failed to trigger runs'
-        )
-      ];
-      return this.failures;
-    }
+    if (runs != undefined){
```

Comment:
this diff is quite hard to interpret, but I think that what is happening is that you have wrapped an existing group of statements into an `if(cond) {unusual}else{prexisting}` block. Better use the return-early idiom: `if(cond) {unusual; return}`. Its not just the diff, it reduces cognitive load - it means from that point on, the reader knows for sure that the unusual "cond" condition is NOT true. With the if/else, you have to remember it is still true, and that code later on after the else condition may still have to deal with it.